grc/gui/Application.py

```diff
@@ -542,8 +542,12 @@ def flow_graph_update(fg=flow_graph):
                 while response == Gtk.ResponseType.APPLY:  # rerun the dialog if Apply was hit
                     response = self.dialog.run()
                     if response in (Gtk.ResponseType.APPLY, Gtk.ResponseType.ACCEPT):
-                        flow_graph_update()
                         page.state_cache.save_new_state(flow_graph.export_data())
+                        ### Following  lines force an complete update of io ports
```

Comment:
Updating all ports on param modification is, at the very least, a safe choice! Nice, thank you.

One question: Why did this migrate from before to after the state saving?

Reply:
I just immated the action after 'quit' as this was the only way to update all ports. Only moving the flow_graph_update 
behind
    page.state_cache.save_new_state(flow_graph.export_data())

didn't make it.
But the combination of the three lines did it !